### DISCUSSION TITLE
subapaseClient data is returning null

### DISCUSSION THREAD

**Discussion by prtpj1** [supabase/supabase#12089] [Questions] [answered]
Hi, I'm learning and trying code with supabase.
Its a simple chat project to praticing.

My issue is with `const { data } = await supabaseClient` in try/catch.
data is returning null and I dont know why.

I type a new msg on chat, browser console give me this erro:

> data null
chat.js?4cc9:75 error TypeError: Cannot read properties of null (reading '0')
    at newChatMsg (chat.js?4cc9:71:21)

But even with this error, all new msgs were inserted in supabase table "messages". With a refresh page, the chat show all msgs, including the last typed

This is part of my code:
```
export default function ChatPage() {
    const router = useRouter();
    const user = router.query.username;
    const [chatMessage, setChatMessage] = React.useState('');
    const [chatListMsg, setChatListMsg] = React.useState([]);

    React.useEffect(() => {
        supabaseClient
            .from('messages')
            .select('*')
            .order('id', { ascending: false })
            .then(({ data }) => {
                setChatListMsg(data);
            });
    }, []);

const newChatMsg = async (msg) => {
        const date = new Date().toLocaleDateString(
            'pt-BR',
            { day: '2-digit', month: '2-digit', year: 'numeric' });
        const time = new Date().toLocaleTimeString({ hour: '2-digit', minute: '2-digit' });
        const dateTime = `${date} - ${time}`;

        const newMessage = {
            from: user,
            text: msg,
            time: dateTime,
        };
        try {
            const { data } = await supabaseClient
                .from('messages')
                .insert([newMessage]);
            console.log('🚀  data', data);

            setChatListMsg([
                data[0],
                ...chatListMsg,
            ]);
        } catch (error) {
            console.log('error', error);
        }

        setChatMessage('');
    };
```

**Comment by GaryAustin1** [comment, marked as answer]
insert() does not return data without adding a .select() to it.  So data is coming back null.

**Reply by prtpj1** [reply]
This way?
```
try {
            const { data } = await supabaseClient
                .from('messages')
                .select('*')
                .insert([newMessage]);
```

**Reply by GaryAustin1** [reply]
I'm not sure if the order matters, but it is shown as
```
const { data } = await supabaseClient
                .from('messages')
                .insert([newMessage]);
                .select('*')
```
In the js documents.
![image](https://user-images.githubusercontent.com/54564956/216152414-7c5b6f9a-b9df-4fdb-af8e-bb2f734b2623.png)

**Reply by prtpj1** [reply]
Gary, thx too much. I'm trying to figure out how to make its work a few days, I did ask to some friends, we tried many things, I thought could be a simple thing...and BANG...you gave me the simple thing...
